googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1147,6 +1147,46 @@ class NativeArray {
     GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__): \
       fail(gtest_msg.value)
 
+#define GTEST_TEST_THROW_WHAT_(statement, expected_exception, regex, fail) \
+  GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
+  if (::testing::internal::ConstCharPtr gtest_msg = "") { \
+    bool gtest_caught_expected = false; \
+    try { \
+      GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement); \
+    } \
+    catch (expected_exception const& e) { \
+      const ::testing::internal::RE& gtest_regex = (regex); \
+      const bool matched = ::testing::internal::RE::PartialMatch(::testing::PrintToString(e), gtest_regex); \
```

Comment:
On Saturday, December 5, 2015, Caleb Morse <notifications@github.com> wrote:

> In googletest/include/gtest/internal/gtest-internal.h
> <https://github.com/google/googletest/pull/641#discussion_r46762520>:
>
> > @@ -1147,6 +1147,46 @@ class NativeArray {
> >      GTEST_CONCAT_TOKEN_(gtest_label_testthrow_, __LINE__): \
> >        fail(gtest_msg.value)
> >
> > +#define GTEST_TEST_THROW_WHAT_(statement, expected_exception, regex, fail) \
> > +  GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
> > +  if (::testing::internal::ConstCharPtr gtest_msg = "") { \
> > +    bool gtest_caught_expected = false; \
> > +    try { \
> > +      GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement); \
> > +    } \
> > +    catch (expected_exception const& e) { \
> > +      const ::testing::internal::RE& gtest_regex = (regex); \
> > +      const bool matched = ::testing::internal::RE::PartialMatch(::testing::PrintToString(e), gtest_regex); \
>
> How can it be more like EXPECT_THAT? As far as I can tell, Matcher's are
> part of google mock and haven't been ported to google test yet.
>
Then make this macro part of gmock?

The split is a lame historical accident and the goal is unification, btw.



> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/googletest/pull/641/files#r46762520>.
>